src/dominion/player.rs

```diff
@@ -191,10 +234,81 @@ impl Player {
 #[cfg(test)]
 mod tests {
     use super::*;
+    use crate::dominion::KingdomSet;
+
+    fn create_player() -> Player {
+        let supply = Rc::new(RefCell::new(Supply::new(KingdomSet::FirstGame.cards(), 2)));
+
+        Player::new(supply)
+    }
+
+    #[test]
+    fn buy_card_copper() {
+        let mut p = create_player();
+
+        p.phase = Some(TurnPhase::Buy {
+            remaining_buys: 1,
+            total_wealth: 0,
+        });
+
+        let r = p.buy_card(&CardKind::Copper);
+
+        assert!(r.is_ok());
+        assert_eq!(p.discard_pile[0], &CardKind::Copper);
```

Comment:
Assert that the supply of coppers have decreased by one?

Reply:
done